frameworks/kafka/docs/upgrade.md

```diff
@@ -0,0 +1,92 @@
+---
+post_title: Upgrade Kafka from 0.10.2.1 to 0.11.0.0
+menu_order: 10
+feature_maturity: preview
+enterprise: 'no'
+---
+
+# Upgrade Kafka from 0.10.2.1 to 0.11.0.0
+
+This document explains upgrading the DC/OS Kafka service from 0.10.2.x to 0.11.0.0 version. Please follow the Kafka [upgrade](https://kafka.apache.org/documentation/#upgrade) documentation for more information. Information about the new message format in 0.11.0 is available [here](https://kafka.apache.org/documentation/#upgrade_11_message_format). Kafka 0.11.0 introduces idempotent and transactional capabilies. These new 0.11.0 features requires the new messaging format, and so will not work on older formats. Please also note that 0.11.0 clients can communicate with older brokers, such as 0.10.2 brokers.  
+
+The rolling upgrade procedure (from 0.10.2.x to 0.11.0.0) is explained below. This is the recommended way of upgrading to 0.11.0.0 in order to guarantee no downtime while upgrading Kafka.
+
+
+1. Upgrade the code to the new version:
+   * Upgrade Kafka software to 0.11.0.0.
+   * Restart the brokers one at a time.    
+   * Wait until the entire cluster is upgraded.
+
+2. Change inter.broker.protocol.version to 0.11.0.0:
+
+   * Edit server.properties files and change inter.broker.protocol.version to 0.11.0.0.
```

Comment:
This explains the rolling upgrade process for no downtime, described in https://kafka.apache.org/documentation/#upgrade for 0.10.2.x to 0.11.0.0